components/script/dom/response.rs

```diff
@@ -441,11 +445,24 @@ impl Response {
         }
     }
 
+    pub fn set_stream_consumer(&self, sc: Option<StreamConsumer>) {
+        *self.stream_consumer.borrow_mut() = sc;
+    }
+
+    pub fn stream_chunk(&self, stream: &[u8]) {
+        if let Some(stream_consumer) = self.stream_consumer.borrow_mut().take() {
```

Comment:
take() consumes the value, so we want  as_ref() instead.

Reply:
Done